pkg/executor/wasm/executor.go

```diff
@@ -174,78 +174,59 @@ func (e *Executor) Run(ctx context.Context, executionID string, job model.Job, j
 	engine := tracedRuntime{wazero.NewRuntimeWithConfig(ctx, engineConfig)}
 	defer closer.ContextCloserWithLogOnError(ctx, "engine", engine)
 
-	inputVolumes, err := storage.ParallelPrepareStorage(ctx, e.StorageProvider, job.Spec.Inputs)
-	if err != nil {
-		return nil, err
-	}
-	defer func() {
-		log.Ctx(ctx).Debug().
-			Str("Execution", executionID).
-			Msg("attempting cleanup of inputs for execution")
-		err := storage.ParallelCleanStorage(ctx, e.StorageProvider, inputVolumes)
-		if err != nil {
-			log.Ctx(ctx).Error().
-				Err(err).
-				Str("Execution", executionID).
-				Msg("errors occurred when cleaning up inputs")
-		}
-	}()
-
-	rootFs, err := e.makeFsFromStorage(ctx, jobResultsDir, inputVolumes, job.Spec.Outputs)
+	rootFs, err := e.makeFsFromStorage(ctx, args.ResultsDir, args.Inputs, args.Outputs)
 	if err != nil {
 		return executor.FailResult(err)
 	}
 
 	// Create a new log manager and obtain some writers that we can pass to the wasm
 	// configuration
-	logs, err := wasmlogs.NewLogManager(ctx, executionID)
+	logs, err := wasmlogs.NewLogManager(ctx, args.ExecutionID)
 	if err != nil {
 		return executor.FailResult(err)
 	}
 	stdout, stderr := logs.GetWriters()
 
 	// Store the LogManager for the lifetime of the execution, making sure to tidy up
 	// once complete.
-	e.logManagers.Put(executionID, logs)
+	e.logManagers.Put(args.ExecutionID, logs)
 	defer func() {
-		log.Ctx(ctx).Debug().Str("Execution", executionID).Msg("cleaning up logmanager for execution")
+		log.Ctx(ctx).Debug().Str("Execution", args.ExecutionID).Msg("cleaning up logmanager for execution")
 		logs.Close()
-		e.logManagers.Delete(executionID)
-		log.Ctx(ctx).Debug().Str("Execution", executionID).Msg("logmanager being removed")
+		e.logManagers.Delete(args.ExecutionID)
+		log.Ctx(ctx).Debug().Str("Execution", args.ExecutionID).Msg("logmanager being removed")
 	}()
 
 	// Configure the modules. We don't want to execute any start functions
 	// automatically as we will do it manually later. Finally, add the
 	// filesystem which contains our input and output.
 
-	args := append([]string{job.Spec.Wasm.EntryModule.Name}, job.Spec.Wasm.Parameters...)
-
 	config := wazero.NewModuleConfig().
 		WithStartFunctions().
 		WithStdout(stdout).
 		WithStderr(stderr).
-		WithArgs(args...).
+		WithArgs(append([]string{wasmArgs.EntryModule.Spec.Name}, wasmArgs.Parameters...)...).
```

Comment:
Not a big deal, but I think I preferred the old way because this is quite dense in the middle of this builder.